server/src/main/java/org/apache/iotdb/db/engine/storagegroup/timeindex/FileTimeIndex.java

```diff
@@ -114,8 +81,14 @@ public void close() {
   }
 
   @Override
-  public Set<String> getDevices() {
-    return devices;
+  public Set<String> getDevices(String tsFilePath) {
+    try {
+      TsFileSequenceReader fileReader = FileReaderManager.getInstance().get(tsFilePath, true);
+      return new HashSet<>(fileReader.getAllDevices());
+    } catch (IOException e) {
+      logger.error("Can't read file {} from disk ", tsFilePath, e);
```

Comment:
It's better to throw IOException

Reply:
I will create another PR to throw IOException for the method and all method callers.